### PR TITLE
Initialize redux

### PR DESCRIPTION
**Pull request by ZawadiSumba66**
In this milestone, we initialized redux using store to manage our state, reducers, and actions. We also created the BookList component to display our books and the BookForm for adding new books.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable */
```
Comment (addod19) [inline src/components/Book.js]: Kindly avoid these inline disable linter rules. Kindly remove it

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        BOOK_CATEGORIES.map((value) => (
+          <option value={value} key={value}>
+            {value}
+          </option>
+        ))
+      }
+      {/* <option value="grapefruit">Grapefruit</option>
+                <option value="lime">Lime</option>
+                <option selected value="coconut">Coconut</option>
+                <option value="mango">Mango</option> */}
```
Comment (addod19) [inline src/containers/BooksForm.js]: Kindly remove this comment, use git history instead

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Book from '../components/Book';
+/* eslint-disable react/prop-types */
```
Comment (addod19) [inline src/containers/BooksList.js]: Kindly avoid inline disabling rules

File: src/reducers/books.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          {
+            id: Date.now(),
+            title: action.book.title,
+            category: action.book.category,
+          },
+        ],
+      };
+
+    case REMOVE_BOOK:
+      // let book_to_remove = state.books.filter((book)=> book.id == )
```
Comment (addod19) [inline src/reducers/books.js]: Kindly remove comment, use git history instead

File: src/containers/BooksForm.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const BOOK_CATEGORIES = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+const submitForm = (e) => {
+  e.preventDefault();
+  console.log('Form submitted');
+  console.log(e);
```
Comment (addod19) [inline src/containers/BooksForm.js]: [Optional]
- Let us avoid console.log statements, it slows the application. Kindly create a notification component and render it instead

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @ZawadiSumba66 and @AymanJabr, great job on this milestone so far :muscle: Great implementation of the BookList container :100: Awesome. Right linter configuration is used :1st_place_medal: 

- Kindly copy the contents of `src/App.js` into `src/components/App.js` and remove the `src/App.js` file
- Let us kindly add example books to the initial state object
- `src/reducers/index.js` combines app-specific reducers. **HINT** This is where combine reducers come in

[Optional]
- Kindly move the src/reducers/store to src/. So it becomes src/store


Once these are resolved, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by AymanJabr** [comment]
## We added the Initial state of the books in the reducers/book.js file, as it is the place where the reducer is first initialized.

File: src/reducers/books.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { CREATE_BOOK, REMOVE_BOOK } from '../actions/index';
+
+const initialState = {
+  books: [
+    { id: Math.random(), title: 'I am Malala', category: 'Action' },
+    { id: Math.random(), title: 'The Survivors', category: 'Horror' },
+    { id: Math.random(), title: 'Ghost Rider', category: 'History' },
+    { id: Math.random(), title: 'Ghost Rider', category: 'Biography' },
+  ],
+};
```
Comment (bellom) [inline src/reducers/books.js]: Good work.

Yes, the reducer is first initialized here but in React you can always access the state anywhere in your code, keep in mind reducer is for handling actions.

Kindly improve your code by creating the book example in `src.index.js`

From Pathwright;
image

Let's keep to the instructions given on pathwright to reduce time spent on creating CRR, and also we both know it is mandatory for all students.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by mimipeshy** [comment]
## STATUS: APPROVED 🥳 👯 
Hello team @ZawadiSumba66 👋🏾 👋🏾 congratulations on making it this far in this milestone 🥇 

#### Highlights
- You have worked on all changes requested by the previous TSE ✔️ 

With that, I approve of this project 😄 

Regards
__Peris Ndanu __
[GitHub ](https://github.com/mimipeshy)| [LinkedIn ](https://www.linkedin.com/in/peris-ndanu/)| [Twitter](https://twitter.com/pygirl254)

**Pull request merged**